### ISSUE TITLE
card payment failed " An error occured while attempting to decrypt some parameters"

### ISSUE THREAD

**Issue by HetalGohil** [wilburt/rave_flutter#50]
when i tried to do card payment or mpesa it is giving error 
and console i m getting this 
*** DioError ***:
I/flutter (26380): uri: https://ravesandboxapi.flutterwave.com/flwv3-pug/getpaidx/api/charge
I/flutter (26380): DioError [DioErrorType.RESPONSE]: Http status error [400]
I/flutter (26380): uri: https://ravesandboxapi.flutterwave.com/flwv3-pug/getpaidx/api/charge
I/flutter (26380): statusCode: 400
I/flutter (26380): headers:
I/flutter (26380):  connection: keep-alive
I/flutter (26380):  x-powered-by: Express
I/flutter (26380):  access-control-allow-headers: Origin, X-Requested-With, Content-Type, Accept, Authorization, v3-xapp-id, flw-auth-token, mra-auth-token, alt_mode_auth, mid
I/flutter (26380):  date: Mon, 01 Feb 2021 07:28:47 GMT
I/flutter (26380):  x-moesif-transaction-id: 1b223e84-06c8-4a4c-ba13-feac39de9c40
I/flutter (26380):  access-control-allow-origin: *
I/flutter (26380):  access-control-allow-methods: PUT, POST, GET, DELETE, OPTIONS
I/flutter (26380):  content-length: 103
I/flutter (26380):  etag: W/"67-zshR+cq5UdFzzywHJM9Wlg"
I/flutter (26380):  via: 1.1 vegur
I/flutter (26380):  content-type: application/json; charset=utf-8
I/flutter (26380):  server: Cowboy
I/flutter (26380): Response Text:
I/flutter (26380): {"status":"error","message":"An error occured while attempting to decrypt some parameters","data":null}
I/flutter (26380): RaveResult{status: RaveStatus.error, rawResponse: null, message: An error occured while attempting to decrypt some parameters}

**Comment by OTopman** [comment]
Have you gotten the solution??

**Comment by Ekeminie** [comment]
any solutions to this?

**Comment by sudo-which-qp** [comment]
None yet... I'm even trying to see why this is happening if it's coming from Flutterwave... it might be

**Comment by sudo-which-qp** [comment]
<img width="798" alt="Screenshot 2022-05-17 at 4 02 45 PM" src="https://user-images.githubusercontent.com/44076215/168844303-f3c40847-b130-4e31-b9e9-d597a529f9e0.png">

this is the reason for the error... It's from Flutterwave

**Comment by Ekeminie** [comment]
> <img alt="Screenshot 2022-05-17 at 4 02 45 PM" width="798" src="https://user-images.githubusercontent.com/44076215/168844303-f3c40847-b130-4e31-b9e9-d597a529f9e0.png">
> 
> this is the reason for the error... It's from Flutterwave

True.
But that's not enough info though.

**Comment by sudo-which-qp** [comment]
> > <img alt="Screenshot 2022-05-17 at 4 02 45 PM" width="798" src="https://user-images.githubusercontent.com/44076215/168844303-f3c40847-b130-4e31-b9e9-d597a529f9e0.png">
> > this is the reason for the error... It's from Flutterwave
> 
> True. But that's not enough info though.

I'm still trying to know why this is happening... if you have seen the reason after now please do well to tell me... this issue is till disturbing the flutterwave main plugin it self...

**Comment by Ekeminie** [comment]
Checking..I'll post an update once I figure this

**Comment by sudo-which-qp** [comment]
> Checking..I'll post an update once I figure this

It seems Flutterwave does not require the use of the Encryption key anymore to validate the transaction... I found another Plugin that is owned by the FlutterWave team

https://github.com/Flutterwave/flutter_standard

flutterwave_standard: 1.0.2

Check it out... it's not all great but you customize it to your taste

**Comment by Ekeminie** [comment]
Okay?

> > Checking..I'll post an update once I figure this
> 
> It seems Flutterwave does not require the use of the Encryption key anymore to validate the transaction... I found another Plugin that is owned by the FlutterWave team
> 
> https://github.com/Flutterwave/flutter_standard
> 
> flutterwave_standard: 1.0.2
> 
> Check it out... it's not all great but you customize it to your taste

Thanks. I'll try seeing if I can fix the other plugins using this discovery

**Comment by OTopman** [comment]
> > > <img alt="Screenshot 2022-05-17 at 4 02 45 PM" width="798" src="https://user-images.githubusercontent.com/44076215/168844303-f3c40847-b130-4e31-b9e9-d597a529f9e0.png">
> > > this is the reason for the error... It's from Flutterwave
> > 
> > 
> > True. But that's not enough info though.
> 
> I'm still trying to know why this is happening... if you have seen the reason after now please do well to tell me... this issue is till disturbing the flutterwave main plugin it self...

It's the encryption method you used

**Comment by Ekeminie** [comment]
> > > > <img alt="Screenshot 2022-05-17 at 4 02 45 PM" width="798" src="https://user-images.githubusercontent.com/44076215/168844303-f3c40847-b130-4e31-b9e9-d597a529f9e0.png">
> > > > this is the reason for the error... It's from Flutterwave
> > > 
> > > 
> > > True. But that's not enough info though.
> > 
> > 
> > I'm still trying to know why this is happening... if you have seen the reason after now please do well to tell me... this issue is till disturbing the flutterwave main plugin it self...
> 
> It's the encryption method you used

care to explain a bit more?

**Comment by sudo-which-qp** [comment]
> > > > <img alt="Screenshot 2022-05-17 at 4 02 45 PM" width="798" src="https://user-images.githubusercontent.com/44076215/168844303-f3c40847-b130-4e31-b9e9-d597a529f9e0.png">
> > > > this is the reason for the error... It's from Flutterwave
> > > 
> > > 
> > > True. But that's not enough info though.
> > 
> > 
> > I'm still trying to know why this is happening... if you have seen the reason after now please do well to tell me... this issue is till disturbing the flutterwave main plugin it self...
> 
> It's the encryption method you used

There is no other encryption method provided than you putting your encryption key from Flutterwave into the Plugin, but as it seems, Flutterwave might have changed some details on how people communicate with their API for transactions... because I tried using 3 different FlutterWave Plugin and they all returned the same error... and I have used those Plugins in the past and I didn't get any issue like this.

As I said before I found out on their Official Documentation Page for all Platform SDK that they have created another Plugin for Payments. This is the GitHub Repo for it: https://github.com/Flutterwave/flutter_standard... Read through it.

I'm guessing since they move all their API's to FlutterWave 3.0 in the previous months coming that's when they made those changes...

But again I might just be wrong.
